packager/file/udp_file.cc

```diff
@@ -163,27 +246,31 @@ bool UdpFile::Open() {
   if (!options)
     return false;
 
-  ScopedSocket new_socket(socket(AF_INET, SOCK_DGRAM, 0));
-  if (new_socket.get() == INVALID_SOCKET) {
-    LOG(ERROR) << "Could not allocate socket.";
+  struct sockaddr_in local_sock_addr = {0};
+
+  struct addrinfo hints = {0};
+  struct hostent *host_info = gethostbyaddr(options->address().c_str(), (socklen_t)sizeof(SOCKET), AF_UNSPEC);
+  struct addrinfo *res;
+
+  SOCKET s;
+
+  hints.ai_family = AF_UNSPEC;
+  hints.ai_socktype = SOCK_DGRAM;
+
+  if (getaddrinfo(host_info->h_name, std::to_string(options->port()).c_str(), &hints, &res) != 0) {
+    LOG(ERROR) << "getaddrinfo";
     return false;
   }
-
-  struct in_addr local_in_addr = {0};
-  if (inet_pton(AF_INET, options->address().c_str(), &local_in_addr) != 1) {
-    LOG(ERROR) << "Malformed IPv4 address " << options->address();
+  if ((s = socket(res->ai_family, res->ai_socktype, res->ai_protocol)) == -1) {
```

Comment:
You can do ScopedSocket new_socket(socket(...)). There is no need to check for the result before assigning it to ScopedSocket.